### PR TITLE
scripts: add generate-release-notes scripts

### PR DESCRIPTION
**Pull request by BethGriggs**
These scripts are used to generate release notes - detailed usage information is included in the README.md. The scripts have been migrated from https://github.com/BethGriggs/release-notes-prototype on the commit 906420d.

### CODE REVIEW

File: generate-release-notes/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# OpenJDK Release Notes Tool
+
+This repository contains two scripts that can be used to generate OpenJDK notes. Typical usage will be to first generate the list of commits for the release using the `fetchCommitList.js` script and then run `fetchReleaseNotes.js` script, supplying the commit list as an input.
```
Comment (karianna) [inline generate-release-notes/README.md:3]:
```suggestion
This repository contains two scripts that can be used to generate OpenJDK release notes. Typical usage will be to first generate the list of commits for the release using the `fetchCommitList.js` script and then run the `fetchReleaseNotes.js` script, supplying the commit list from `fetchCommitList.js` as an input.
```

File: generate-release-notes/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  {
+    "id": "JDK-8287017",
+    "commit": "25ac222aa2e04f934ffd989e3ee355157f497fde",
+    "title": "8287017: Bump update version for OpenJDK: jdk-11.0.17"
+  },
+  ...
+]
+```
+
+Note that the same commit may reference multiple JDK issues.
```
Comment (karianna) [inline generate-release-notes/README.md:45]:
```suggestion
*NOTE:* The same commit may reference multiple JDK issues.
```

**Review by karianna — CHANGES_REQUESTED** [review]
Changes look largely good but not sure you meant to commit the output folder?

File: generate-release-notes/README.md (hunk 1 above)
Comment (BethGriggs) [reply generate-release-notes/README.md:3]: It was intentional to provide some example output to share. I can delete (or reduce) if it is preferred.

**Review by sxa — APPROVED** [review]

**Comment by BethGriggs** [comment]
@karianna, I believe I've addressed your README comments. I also renamed the `output` directory to be `sample-output` for clarity. I still don't mind removing these - but I have found having some sample-output in the git tree useful to manually verify the impact of script changes, such as adding ProblemLists earlier - https://github.com/BethGriggs/release-notes-prototype/commit/575061ad8d9454079b21d1d5574482b13c1296a5.

I applied the fixes directly in https://github.com/BethGriggs/release-notes-prototype rather than here as I'd like this PR to be a clean switchover from the other repository at a specified commit. My plan is to archive the original repository as soon as this lands to avoid any divergence.

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
@BethGriggs Safe for me to merge this?

**Comment by sxa** [comment]
I'm going to merge this, update https://ci.adoptopenjdk.net/job/create_release_notes/configure to point to this repo, and run a test on it :-)

**Pull request merged**